app/components/LeftPaneEntries/AssistItem/utils.ts

```diff
@@ -1,64 +1,81 @@
 import {
-    listToGroupList,
     listToMap,
-    isDefined,
-    mapToMap,
     randomString,
 } from '@togglecorp/fujs';
 
 import {
-    Matrix1dMappingsItem,
     Matrix1dWidget,
-    Matrix2dMappingsItem,
-    OrganigramMappingsItem,
+    Matrix1dValue,
+    Matrix2dValue,
     Matrix2dWidget,
-    ScaleMappingsItem,
-    ScaleWidget,
-    SelectMappingsItem,
-    SingleSelectWidget,
-    MultiSelectMappingsItem,
-    MultiSelectWidget,
-    OrganigramWidget,
-    GeoLocationWidget,
 } from '#types/newAnalyticalFramework';
-import {
-} from '#types/newEntry';
 
 import {
     getType,
-    DeepReplace,
 } from '#utils/types';
 
-import { GeoArea } from '#components/GeoMultiSelectInput';
 import {
     PartialAttributeType,
 } from '#components/entry/schema';
 
 type Matrix1dWidgetAttribute = getType<PartialAttributeType, { widgetType: 'MATRIX1D' }>;
 type Matrix2dWidgetAttribute = getType<PartialAttributeType, { widgetType: 'MATRIX2D' }>;
+/*
 type ScaleWidgetAttribute = getType<PartialAttributeType, { widgetType: 'SCALE' }>;
 type SingleSelectWidgetAttribute = getType<PartialAttributeType, { widgetType: 'SELECT' }>;
 type MultiSelectWidgetAttribute = getType<PartialAttributeType, { widgetType: 'MULTISELECT' }>;
 type OrganigramWidgetAttribute = getType<PartialAttributeType, { widgetType: 'ORGANIGRAM' }>;
 type GeoLocationWidgetAttribute = getType<PartialAttributeType, { widgetType: 'GEO' }>;
+*/
+
+export function isValidObject(value: unknown): value is Record<string, unknown> {
+    return typeof value === 'object' && !Array.isArray(value) && value !== null && value !== undefined;
+}
 
-export function createMatrix1dAttr(
-    mappings: Matrix1dMappingsItem[] | undefined,
+export function isValidStringArray(value: unknown): value is string[] {
+    const isArray = typeof value === 'object' && Array.isArray(value) && value !== null && value !== undefined;
+    if (!isArray) {
+        return false;
+    }
+    return value.every((item) => typeof item === 'string');
+}
+
+// TODO: Write tests
+export function createMatrix1dAttrFromTags(
+    unsafeMappings: Matrix1dValue | undefined,
     widget: Matrix1dWidget,
 ): Matrix1dWidgetAttribute | undefined {
-    if (!mappings || mappings.length <= 0) {
+    // NOTE: We are supposed to get Matrix1dValue, but adding one more validation layer
+    const mappings = unsafeMappings as unknown;
```

Comment:
The type of `unsafeMappings` is already defined. Why are we retyping it as `unknown`?